server/src/main/scala/org/http4s/server/middleware/CORS.scala

```diff
@@ -54,19 +55,28 @@ object CORS {
         else
           config.exposedHeaders.map(headerFromStrings("Access-Control-Expose-Headers", _))
 
+      def checkVaryHeader(response: Response[G]): Option[Header] = {
+        val maybeHeader = response.headers.get(CaseInsensitiveString("Vary"))
+        if (maybeHeader.isEmpty) Option(Header("Vary", "Origin,Access-Control-Request-Method"))
+        else maybeHeader
```

Comment:
This could be `maybeHeader.getOrElse(Header(...))`.  But I like @jmcardon's idea.

Reply:
I'd like to polish the PR a bit - thanks for the comments